### PR TITLE
Add clang-format 7.0.1 to CI image

### PR DESCRIPTION
**Pull request by palmerlao**
This installs `clang-format` 7.0.1 (it seems to me 7.x.x is considered "stable" LLVM).
```
➜  vowpal_wabbit git:(palmerlao/update-ci-image) ✗ docker run -ti fb9f8e9fdc77 clang-format --version
clang-format version 7.0.1 (tags/RELEASE_701/final)
➜  vowpal_wabbit git:(palmerlao/update-ci-image) ✗ docker run -ti fb9f8e9fdc77 g++ --version
g++ (Ubuntu 4.9.4-2ubuntu1~14.04.1) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

➜  vowpal_wabbit git:(palmerlao/update-ci-image) ✗ docker run -ti fb9f8e9fdc77 cmake --version
cmake version 3.2.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).

```

### CODE REVIEW

**Comment by palmerlao** [comment]
Also, is `cpprestsdk` still used?

**Comment by jackgerrits** [comment]
Yes, this image is also used as a base for the reinforcement_learning repo where cpprestsdk is used.

Traditionally we've targeted older versions of linux to ensure compatibility. I think there will be less resistance to simply installing a newer clang-format on the 14.04 image if that is possible.

**Comment by jackgerrits** [comment]
What is involved in getting the required version of clang-format into the current ubuntu image?

**Comment by JohnLangford** [comment]
Status on this?  It would be nice to avoid needing the latest and greatest to build VW.

**Comment by palmerlao** [comment]
Sorry, haven't made much time for this yet. I do think it is possible and
will try this weekend at the latest. Is this blocking anything other than
the formatting work?

**Comment by JohnLangford** [comment]
It's not blocking, but it would be nice to land a big patch like this
because it keeps getting obsolete as the underlying codebase changes.

-John

File: .pipelines/Dockerfile [hunk 1]
```diff
@@ -1,8 +1,7 @@
 FROM ubuntu:14.04 AS build
 
 # Upgrade cmake to 3.2
-RUN apt-get update
-RUN apt-get install -y software-properties-common python-software-properties debconf-utils
+RUN apt-get update && apt-get install -y software-properties-common python-software-properties debconf-utils
```
Comment (palmerlao) [inline .pipelines/Dockerfile]: I don't understand very well why this change was necessary, but without it, running `docker build - < .pipelines/Dockerfile` gave me an error on this step that looked like `apt-get update` hadn't been run. Combining the two lines made it work again.

The [Docker documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) indicates that this seems to be the preferable way to run `apt-get update`.

**Comment by palmerlao** [comment]
Friendly reminder @JohnLangford

Is this something you all are still interested in? The change is a fair bit smaller now and I think uses the same versions of other software as before.

**Comment by JohnLangford** [comment]
Yep merged, thanks :-) (I was just locked on the COLT deadline until yesterday.)

**Pull request merged**